### PR TITLE
[Mobile] Category budget actions modal

### PR DESCRIPTION
**Pull request by joel-jeremy**
Built on top of https://github.com/actualbudget/actual/pull/2491 (need to merge that before this)

- Added the following budget actions through a modal:
  - Copy last month's budget
  - Set to 3 month average
  - Set to 6 month average
  - Set to yearly average
  - Apply budget template

### CODE REVIEW

**Comment by youngcw** [comment]
I woudln't be against removing the inline edit of the budgeted amount.  I don't add manually add the budget amounts each month, but I would think that a amount edit modal with the other options wouldn't be that much of a slow down at all.

**Comment by joel-jeremy** [comment]
I totally agree with the above assessment. That's why I'm also leaning towards opening the budget menu (modal) when user clicks on the budget instead of it being in the category menu (modal) but I wanted to check what others think first before I implement it. :)

**Comment by youngcw** [comment]
I think this looks good

**Comment by Teprifer** [comment]
+1 to the modal, in a way I like it more because the text can be bigger, and it's a good way to bring those other options in.

My only feedback would be for the phone pop-up keyboard enter key to confirm the value and dismiss the dialog.

File: packages/desktop-client/src/components/mobile/transactions/FocusableAmountInput.tsx [hunk 1]
```diff
@@ -216,7 +273,7 @@ export const FocusableAmountInput = memo(function FocusableAmountInput({
               borderBottomWidth: 1,
               borderColor: '#e0e0e0',
               justifyContent: 'center',
-              transform: [{ translateY: 0.5 }],
+              // transform: [{ translateY: 0.5 }],
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/mobile/transactions/FocusableAmountInput.tsx:276]: **🥜 nitpick:** ‏
```suggestion
```

File: upcoming-release-notes/2501.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+category: Enhancements
+authors: [joel-jeremy]
+---
+
+Mobile budget action menu modals.
```
Comment (MatissJanis) [inline upcoming-release-notes/2501.md:6]: **🥜 nitpick:** ‏you know what my feedback here will be 😄

**Review by MatissJanis — APPROVED** [review]
Overall LGTM!

**Review by MatissJanis — DISMISSED** [review]

**Review by MatissJanis — APPROVED** [review]

**Review by MatissJanis — APPROVED** [review]

**Pull request merged**